sipXproxy/lib/authplugins/EnforceAuthRules.cpp

```diff
@@ -269,16 +269,26 @@ EnforceAuthRules::authorizeAndModify(const UtlString& id,    /**< The authentica
             }
             else
             {
-               result = DENY;
-               Os::Logger::instance().log(FAC_AUTH, PRI_WARNING,
-                             "EnforceAuthRules[%s]::authorizeAndModify "
-                             " call '%s' requires '%s'",
-                             mInstanceName.data(), callId.data(), unmatchedPermissions.data()
-                             );
-               // since the user is at least a valid user, help them debug the configuration
-               // by telling them what permissions would allow this request.
-               reason.append("Requires ");
-               reason.append(unmatchedPermissions);
+              if (!unmatchedPermissions.isNull())
+              {
+                result = DENY;
+                Os::Logger::instance().log(FAC_AUTH, PRI_WARNING,
+                              "EnforceAuthRules[%s]::authorizeAndModify "
+                              " call '%s' requires '%s'",
+                              mInstanceName.data(), callId.data(), unmatchedPermissions.data()
+                              );
+                // since the user is at least a valid user, help them debug the configuration
+                // by telling them what permissions would allow this request.
+                reason.append("Requires ");
+                reason.append(unmatchedPermissions);
+              }
+              else
+              {
+                result = ALLOW;
```

Comment:
Hi, Joegen. Thanks for your comment.
From you description I have some difficulties for understanding what needs to be changed.
I have got that:
1. It is required to return result DENY for all cases where isAuthorized == false
2. We have to check logs to clear why unmatchedPermission == NULL if user not found.
Please confirm if I understand your comment correctly.

Thanks.